kratos/solving_strategies/schemes/residual_based_adjoint_bossak_scheme.h

```diff
@@ -992,23 +992,23 @@ class ResidualBasedAdjointBossakScheme : public Scheme<TSparseSpace, TDenseSpace
     {
         KRATOS_TRY
 
-        const int number_of_elements = rEntityContainer.size();
-
         Vector adjoint2_aux, adjoint3_aux;
-#pragma omp parallel for private(adjoint2_aux, adjoint3_aux)
-        for (int i = 0; i < number_of_elements; ++i) {
-            auto& r_entity = *(rEntityContainer.begin() + i);
+        auto aux_TLS = std::make_pair(adjoint2_aux, adjoint3_aux);
+        block_for_each(rEntityContainer, aux_TLS, [&, this](typename TEntityContainerType::value_type& rEntity, std::pair<Vector,Vector>& rAdjointTLS){
```

Comment:
Just a minor suggestion, 
```suggestion
        using tls_type = std::tuple<Vector, Vector>;
        block_for_each(rEntityContainer, tls_type(), [&, this](typename TEntityContainerType::value_type& rEntity, tls_type& rAdjointTLS){
```